bot/code_review_bot/workflow.py

```diff
@@ -275,8 +276,23 @@ def _in_group(dep_id):
                     stats.report_task(task, task_issues)
                     issues += task_issues
 
-                    for patch in task.build_patches(artifacts):
+                    task_patches = task.build_patches(artifacts)
+                    for patch in task_patches:
                         revision.add_improvement_patch(task.name, patch)
+
+                    # Report a problem when tasks in erroneous state are found
+                    # but no issue or patch has been processed by the bot
```

Comment:
You're not actually checking the state here, are we checking it somewhere else?

Reply:
I'm not checking it because we are already processing an erroneous task at this stage of the workflow.

Reply:
Fixed in 6d7a6e5.